### DISCUSSION TITLE
How to get console into the build.

### DISCUSSION THREAD

**Discussion by rweisman** [ipxe/ipxe#867] [Q&A] [answered]
Ok.. Sure I am missing something,   I uncommented the line in config/general.h
#define CONFIG_CMD              /* Option configuration console */

and also the line in config/console.h
#define CONSOLE_FRAMEBUFFER     /* Graphical framebuffer console */

However , I don't have a console command in my build.  Am I missing something?

Thanks

**Comment by NiKiZe** [comment]
https://ipxe.org/cmd/console#build_options has info about exactly what you need, which you seem to have.

The next part would be to ensure that you are using your new build and not something else. For example verify with embedded script to be sure you are booting the correct image. You could also verify with usb or similar. Just to rule out the build config being wrong in some way.

You could maybe provide the command you are using to build and any maybe some info on your DHCP config?

**Comment by mcb30** [comment]
> Ok.. Sure I am missing something, I uncommented the line in config/general.h #define CONFIG_CMD /* Option configuration console */

Looks like a typo: I think you wanted to enable `CONSOLE_CMD` rather than `CONFIG_CMD`

**Reply by rweisman** [reply]
OK. Will double check that I uncommented the CONSOLE_CMD.  Probably just I cut and pasted the wrong line in my hurry to leave the office.
My dnsmasq config is booting the snponly.efi that I renamed to bootipxe.efi.  Nothing special there.
I use the make bin-x86_64-efi/snponly.efi EMBED=bootipxe.ipxe,   
I was able to add the menu command by un-commenting the menu line in the general.h.   So when the ipxe header boots it show menu in the list of other stuff.  I am @home now, so I will check in the morning.    
Also, I have the latest version from git.  OS I am running to create and pxe boot is AlmaLinux 9.   I am able to get a menu and select Items and have them boot.  I just want to create a more similar look to grub pxe menu's that I was using on my currentl production PXE server.

Thank you all for responding.

**Reply by mcb30** [reply, marked as answer]
`MENU_CMD` (and `CONFIG_CMD`) are both enabled by default in `config/general.h` - are you sure you're cloning an unmodified iPXE tree?

**Reply by rweisman** [reply]
I am pretty sure that I am using the Main iPXE Tree.  I will clone it again into a different folder.
Thank you

**Reply by rweisman** [reply]
Ok.  I cloned the master tree and it now works.

Thanks for the suggestion.

**Reply by mcb30** [reply]
Glad you got it working!
